### ISSUE TITLE
@types/mongodb use `unknown` type instead of TS 2 hacks

### ISSUE THREAD

**Issue by HosseinAgha** [DefinitelyTyped/DefinitelyTyped#38687]
- [X] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [X] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @mrcrane

Use `unknown` type instead of TS 2 hacks (Take a look [here](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/38522#discussion_r328859007)):  

Our `ExtractIdType` alias could use TS 3 `unknown`:
```typescript
type ExtractIdType<TSchema> =
    TSchema extends { _id: infer U } // user has defined a type for _id
    ? (unknown extends U ? ObjectId : U)
    : ObjectId; // user has not defined _id on schema
```

Instead of this: 
```typescript
type ExtractIdType<TSchema> =
    TSchema extends { _id: infer U } // user has defined a type for _id
    ? ({} extends U ? Exclude<U, {}> : U) // Exclude is used here to fix typescript 2 bug when TSchema is "any"
    : ObjectId; // user has not defined _id on schema
```

**Comment by mrcrane** [comment]
I'm assuming that this is dependent on #38686

**Comment by mrcrane** [comment]
This was merged in #38789
